rules-engine/src/rules_engine/engine.py

```diff
@@ -329,6 +329,53 @@ def _initialize_billing_periods(self, billing_periods: List[BillingPeriod]) -> N
         self.bills_summer = []
         self.bills_shoulder = []
 
+
+        # #####
+        # Logic Change:
+        # 'Inclusion' in calculations is now determined by two variables:
+        # billing_period.inclusion_calculated: bool 
+        # billing_period.inclusion_override: bool (False by default)
+        # 
+        # Our logic around the AnalysisType can now disallow
+        # a billing_period from being included or overridden
+        # by marking it as NOT_ALLOWED_IN_CALCULATIONS
+        # 
+        # Options for billing_period.analysis_type:
+        # ALLOWED_HEATING_USAGE = 1 # winter months - allowed in heating usage calculations
+        # ALLOWED_NON_HEATING_USAGE = -1 # summer months - allowed in non-heating usage calculations
+        # NOT_ALLOWED_IN_CALCULATIONS = 0 # shoulder months that fall outside reasonable bounds 
+        # 
+        # Use HDDs to determine if shoulder months
+        # are heating or non-heating or not allowed,
+        # or included or excluded
+        # 
+        # Rough calculations from Steve, this will be ammended:
+        # IF hdds is within 70% or higher of max, allowed
+        # less than 25% of max, not allowed
+        # 
+        # 
+        # IF winter months
```

Comment:
Please see the table in this [Google sheet](https://docs.google.com/spreadsheets/d/1aamBMlOML7tR6OtdZmEZdb_qeVcO5uIb9qDweNYB5R4/edit?usp=sharing). The column headings map to variable names in the pseudo code as follows: Column B maps to analysis_type and column C maps to inclusion_calculated. As mentioned in my previous comment, the table lookup approach should be used for all months, i.e. don't use the hdd-based algorithm for shoulder months.